modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/jobs/VersionMigrationsWorker.kt

```diff
@@ -119,51 +126,20 @@ class VersionMigrationsJob : JobService() {
         }
     }
 
-    @Inject lateinit var podcastManager: PodcastManager
-
-    @Inject lateinit var episodeManager: EpisodeManager
-
-    @Inject lateinit var settings: Settings
-
-    @Inject lateinit var fileStorage: FileStorage
-
-    @Inject lateinit var appDatabase: AppDatabase
-
-    @Inject lateinit var playbackManager: PlaybackManager
-
-    @Volatile private var shouldKeepRunning = true
-
-    override fun onStartJob(jobParameters: JobParameters): Boolean {
-        LogBuffer.i(LogBuffer.TAG_BACKGROUND_TASKS, "VersionMigrationsTask onStartJob")
-
-        Thread {
-            var shouldReschedule = false
-            try {
-                if (!shouldKeepRunning) {
-                    shouldReschedule = true
-                } else {
-                    performMigration()
-                }
-            } catch (t: Throwable) {
-                Timber.e(t)
-                LogBuffer.e(LogBuffer.TAG_BACKGROUND_TASKS, t, "VersionMigrationsTask jobFinished failed.")
-            } finally {
-                LogBuffer.i(LogBuffer.TAG_BACKGROUND_TASKS, "VersionMigrationsTask jobFinished shouldReschedule? $shouldReschedule")
-                jobFinished(jobParameters, shouldReschedule)
-            }
-        }.start()
-
-        return true
-    }
-
-    override fun onStopJob(jobParameters: JobParameters): Boolean {
-        LogBuffer.i(LogBuffer.TAG_BACKGROUND_TASKS, "VersionMigrationsTask onStopJob")
-        shouldKeepRunning = false
-
-        return false
+    override suspend fun doWork(): Result = withContext(Dispatchers.IO) {
```

Comment:
Changing the context to `Dispatchers.IO` as there are file deletion operations during migrations.